File: OBJC/InitStuff/InitStuff/Person.m
//
//  Person.m
//  InitStuff
//
//  Created by Jacob Landman on 1/30/17.
//  Copyright © 2017 Jacob Landman. All rights reserved.
//

#import "Person.h"

@implementation Person

- (id)initWithFirstName:(NSString*)first lastName:(NSString*)last {
    self = [super init];
    self.firstName = first;
    self.lastName = last;
    return self;
}

- (id)initWithAge:(NSInteger) age {
    
    self = [super init];
    return self;
}

- (void)printName {
    NSLog(@"%@ %@", self.firstName, self.lastName);
}

@end

File: mvc-test/mvc-test/Person.swift
//
//  File.swift
//  mvc-test
//
//  Created by Jacob Landman on 1/11/17.
//  Copyright © 2017 Jacob Landman. All rights reserved.
//

import Foundation

class Person {
    private var _firstName: String!
    private var _lastName: String!
    
    var firstName: String {
        get {
           return _firstName
        }
        
        set {
            _firstName = newValue
        }
        
    }
    
    var lastName: String {
        get {
            return _lastName
        }
        
        set {
            _lastName = newValue
        }
        
    }
    
    init(first: String, last: String) {
        self._firstName = first
        self._lastName = last
    }
    
    var fullName: String {
        return "\(_firstName!) \(_lastName!)"
    }
    
}
